pennylane/transforms/core/transform_program.py

```diff
@@ -109,3 +109,55 @@ def is_empty(self):
             bool: Boolean, True if empty, False otherwise.
         """
         return len(self) == 0
+
+    def is_informative(self):
+        """Check if the transform program is informative or not.
+
+        Returns:
+            bool: Boolean, True if empty, False otherwise.
+        """
+        return self.get_last().is_informative
+
+    def __call__(self, tapes):
+        processing_fns_list = []
+        classical_cotransforms_list = []
+
+        for transform_container in self:
+            num_tapes = len(tapes)
+            transform, args, kwargs, cotransform, _ = transform_container
+
+            execution_tapes = []
+            fns = []
+
+            for tape in tapes:
+                new_tapes, fn = transform(tape, *args, **kwargs)
+                execution_tapes.extend(new_tapes)
+                fns.append(fn)
+
+            new_num_tapes = len(new_tapes)
+
+            # Merge the processing function into in a single one
+            def processing_fn(
+                res,
+                num_tapes=num_tapes,
+                new_num_tapes=new_num_tapes,
+                p_fns=tuple(fns),
+            ):  # pylint: disable=cell-var-from-loop
+                final_results = [
+                    p_fns[idx](res[idx * new_num_tapes : (idx + 1) * new_num_tapes])
+                    for idx in range(num_tapes)
+                ]
+                return final_results
+
```

Comment:
Reusing variable names from outer scope inside a local function could make this function more difficult to read and track.  Does `new_num_tapes` refer to the local variable passed into `processing_fn`, or does it refer to the closure variable?  The pylint warning `cell-var-from-loop` seems to arise due to this confusion.

Since this function is not actually relying on closure, we could extract this function to the top of the file as 

```
def combine_processing_fns(results: ResultBatch,
         individual_processing_fns: Callable[[ResultBatch], Result]) -> ResultBatch:
```
We should be able to infer `num_input_tapes` and `num_output_tapes` from the length of `results` and `individual_processing_fns.

And then use `functools.partial` inside the loop:
```
processing_fn = partial(combine_processing_fns, individual_processing_fns=fns)
```

Extracting this code into a helper function will allow us to unit test it.